File: src/dataframe/rust/stats/distributions/t_distribution.test.R
#!/usr/bin/env Rscript

# t distribution test script
# Usage: Rscript t_distribution.test.R <function> <x_or_p> <df> [lower_tail] [log_p]

source("test-helpers.R")

# Define t distribution functions
t_density <- function(x, df, log = FALSE) {
  dt(x, df, log = log)
}

t_cumulative <- function(x, df, lower.tail = TRUE, log.p = FALSE) {
  pt(x, df, lower.tail = lower.tail, log.p = log.p)
}

t_quantile <- function(p, df, lower.tail = TRUE, log.p = FALSE) {
  qt(p, df, lower.tail = lower.tail, log.p = log.p)
}

t_random <- function(n, df) {
  rt(n, df)
}

# Run the test
run_distribution_test(t_density, t_cumulative, t_quantile, t_random)

File: src/dataframe/rust/stats/distributions/t_distribution.test.ts
#!/usr/bin/env -S deno run --allow-read --allow-run

import { callR, printResults, runComparison } from "./test-helpers.ts";

async function main() {
  console.log("Testing R vs Rust t distribution functions...\n");

  const testCases = [
    { func: "dt", distribution: "t_distribution", args: [0, 5, 0] },
    { func: "pt", distribution: "t_distribution", args: [0, 5, 1, 0] },
    { func: "qt", distribution: "t_distribution", args: [0.5, 5, 1, 0] },
  ];

  const results = await runComparison(testCases);
  printResults(results);

  // Test rt separately since it's not available in Rust interface
  console.log("\nR-only tests:");
  try {
    const rtResult = await callR("rt", "t_distribution", 1, 5);
    console.log(`rt(5) = ${rtResult}`);
  } catch (error) {
    console.log(`rt: Error - ${(error as Error).message}`);
  }
}

if (import.meta.main) {
  main().catch(console.error);
}
